saleor/order/models.py

```diff
@@ -59,9 +61,12 @@ def ready_to_fulfill(self):
         fulfilled).
         """
         statuses = {OrderStatus.UNFULFILLED, OrderStatus.PARTIALLY_FULFILLED}
-        qs = self.filter(status__in=statuses, payments__is_active=True)
+        payments = Payment.objects.filter(is_active=True).values("id")
+        qs = self.annotate(
+            matching=Exists(payments.filter(order_id=OuterRef("id")))
+        ).filter(matching=True)
```

Comment:
You should be able to use the expression directly in the filter without assigning it to a new field with `annotate`, like so: `.filter(Exists(...))`. This makes it even faster as the database does not need to create that temporary result column.